# Uber and Lyft Price Prediction - Linear Regression

# Introduction

Author: samueltatsu <br>
Objectives: Developing a linear regression model to predict ride-hailing prices based on relevant factors. This project aims to create an accurate and efficient pricing model, leveraging key variables such as distance, service category, and other relevant parameters. <br>

Dataset: We will be using [Uber and Lyft Dataset Boston, MA](https://www.kaggle.com/datasets/brllrb/uber-and-lyft-dataset-boston-ma) from Kaggle. <br><br>

Brief Method Explanation: <br>
**Linear Regression**
Linear regression is a statistical method used for modeling the relationship between a dependent variable and one or more independent variables by fitting a linear equation to the observed data. The goal is to find the best-fitting line that minimizes the sum of the squared differences between the observed values and the values predicted by the model. Linear regression is widely used for tasks such as predicting sales, analyzing economic trends, and understanding relationships between variables in various fields.

# Import Libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Data loading

In [5]:
data = pd.read_csv('rideshare_kaggle.csv')

Data loaded successfully.

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 693071 entries, 0 to 693070
Data columns (total 57 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   id                           693071 non-null  object 
 1   timestamp                    693071 non-null  float64
 2   hour                         693071 non-null  int64  
 3   day                          693071 non-null  int64  
 4   month                        693071 non-null  int64  
 5   datetime                     693071 non-null  object 
 6   timezone                     693071 non-null  object 
 7   source                       693071 non-null  object 
 8   destination                  693071 non-null  object 
 9   cab_type                     693071 non-null  object 
 10  product_id                   693071 non-null  object 
 11  name                         693071 non-null  object 
 12  price                        637976 non-null  float64
 13 

- 693701 rows data
- 57 total columns <br>
-> 29 float columns <br>
-> 17 int columns <br>
-> 11 object columns

In [9]:
data.describe(include='object').T

count  unique                                   top    freq
id             693071  693071  424553bb-7174-41ea-aeb4-fe06d4f4b9d7       1
datetime       693071   31350                   2018-11-26 23:21:14     156
timezone       693071       1                      America/New_York  693071
source         693071      12                    Financial District   58857
destination    693071      12                    Financial District   58851
cab_type       693071       2                                  Uber  385663
product_id     693071      13  6f72dfc5-27f1-42e8-84db-ccc7a75f6969   55096
name           693071      13                                UberXL   55096
short_summary  693071       9                             Overcast   218895
long_summary   693071      11    Mostly cloudy throughout the day.   202340
icon           693071       7                               cloudy   218895

- timezone column only has 1 unique values across 693071 data which makes this columns useless